### ISSUE TITLE
runwda doesn't work on 14.7.1

### ISSUE THREAD

**Issue by 0x3c3e** [danielpaulus/go-ios#78]
Hi, thanks for the project. I have tried to run it on MacOS 11.5.2, but segfault popped up.
```
➜  /tmp ./ios runwda --bundleid=com.0x3c3e.WebDriverAgentRunner.xctrunner --testrunnerbundleid=com.0x3c3e.WebDriverAgentRunner.xctrunner --xctestconfig=WebDriverAgentRunner.xctest -t --nojson
INFO[0000] Set Trace mode
DEBU[0000] map[--arg:[] --basedir:<nil> --binary:false --bundleid:com.0x3c3e.WebDriverAgentRunner.xctrunner --details:false --env:[] --help:false --nojson:true --output:<nil> --p12file:<nil> --password:<nil> --path:<nil> --pid:<nil> --plist:false --process:<nil> --setlang:<nil> --setlocale:<nil> --stop-at-entry:false --system:false --testrunnerbundleid:com.0x3c3e.WebDriverAgentRunner.xctrunner --trace:true --udid:<nil> --verbose:false --version:false --xctestconfig:WebDriverAgentRunner.xctest <app_path>:<nil> <bundleID>:<nil> <hostPort>:<nil> <key>:[] <profileId>:<nil> <profileTypeId>:<nil> <targetPort>:<nil> apps:false auto:false ax:false date:false debug:false devicename:false devicestate:false diagnostics:false dproxy:false enable:false forward:false image:false info:false install:false kill:false lang:false launch:false list:false listen:false mobilegestalt:false mount:false pair:false pcap:false ps:false readpair:false reboot:false runtest:false runwda:true screenshot:false syslog:false uninstall:false version:false]
DEBU[0000] Looking for device ''
DEBU[0000] Opening connection:0xc00036a700
DEBU[0000] UsbMux sendios.ReadDevicesType on 0xc00000e9b0
DEBU[0000] UsbMux Receive on 0xc00000e9f8
DEBU[0000] no udid specified using default               udid=xx
DEBU[0000] Opening connection:0xc00036aa50
DEBU[0000] UsbMux sendios.ReadPair on 0xc00000f190
DEBU[0000] UsbMux Receive on 0xc00000f1d8
DEBU[0000] UsbMux sendios.connectMessage on 0xc00000f190
DEBU[0000] UsbMux Receive on 0xc00000f928
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc000353480 and wrap with tlsConn0xc000331460
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=true Port=50391 Request=StartService Service=com.apple.mobile.diagnostics_relay
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000353480
DEBU[0000] Opening connection:0xc00036b3c0
DEBU[0000] UsbMux sendios.ReadPair on 0xc0003e42d8
DEBU[0000] UsbMux Receive on 0xc0003e4308
DEBU[0000] Opening connection:0xc00036b560
DEBU[0000] UsbMux sendios.connectMessage on 0xc0003e4a10
DEBU[0000] UsbMux Receive on 0xc0003e4a40
DEBU[0000] enable session ssl on0xc0003c9360 and wrap with tlsConn0xc000331480
INFO[0000] Running wda                                   bundleid=com.0x3c3e.WebDriverAgentRunner.xctrunner testbundleid=com.0x3c3e.WebDriverAgentRunner.xctrunner xctestconfig=WebDriverAgentRunner.xctest
DEBU[0000] Opening connection:0xc000110000
DEBU[0000] UsbMux sendios.ReadPair on 0xc000126020
DEBU[0000] UsbMux Receive on 0xc000126050
DEBU[0000] UsbMux sendios.connectMessage on 0xc000126020
DEBU[0000] UsbMux Receive on 0xc000126758
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc00010e000 and wrap with tlsConn0xc000116018
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=true Port=50394 Request=StartService Service=com.apple.testmanagerd.lockdown.secure
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc00010e000
DEBU[0000] Opening connection:0xc00042a1a0
DEBU[0000] UsbMux sendios.ReadPair on 0xc0004a2350
DEBU[0000] UsbMux Receive on 0xc0004a2380
DEBU[0000] Opening connection:0xc00042a340
DEBU[0000] UsbMux sendios.connectMessage on 0xc0004a2a88
DEBU[0000] UsbMux Receive on 0xc0004a2ab8
DEBU[0000] enable session ssl on0xc000080bc0 and wrap with tlsConn0xc000330010
DEBU[0000] Opening connection:0xc00042a410
DEBU[0000] UsbMux sendios.ReadPair on 0xc0004a2b18
DEBU[0000] UsbMux Receive on 0xc0004a2b48
DEBU[0000] UsbMux sendios.connectMessage on 0xc0004a2b18
DEBU[0000] UsbMux Receive on 0xc0004a3250
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] Dispatching:_notifyOfPublishedCapabilities:
DEBU[0000] Global Dispatcher Received: [_notifyOfPublishedCapabilities:] [{t:binary, v:{"com.apple.private.DTXBlockCompression":2,"com.apple.private.DTXConnection":1}},]
DEBU[0000] enable session ssl on0xc000080c40 and wrap with tlsConn0xc0004c4040
DEBU[0000] Lockdown sendios.valueRequest
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000080c40
DEBU[0000] Opening connection:0xc00042b460
DEBU[0000] UsbMux sendios.ReadPair on 0xc0001c1d18
DEBU[0000] UsbMux Receive on 0xc0001c1d48
DEBU[0000] UsbMux sendios.connectMessage on 0xc0001c1d18
DEBU[0000] UsbMux Receive on 0xc000126a10
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc0001cea40 and wrap with tlsConn0xc0004c4048
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=false Port=50398 Request=StartService Service=com.apple.mobile.installation_proxy
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc0001cea40
DEBU[0000] Opening connection:0xc00036a530
DEBU[0000] UsbMux sendios.ReadPair on 0xc000126db8
DEBU[0000] UsbMux Receive on 0xc000126de8
DEBU[0000] Opening connection:0xc00036a6d0
DEBU[0000] UsbMux sendios.connectMessage on 0xc0001274f0
DEBU[0000] UsbMux Receive on 0xc000127520
DEBU[0000] Lockdown sendmap[string]interface {}
DEBU[0000] Opening connection:0xc0004aee40
DEBU[0000] UsbMux sendios.ReadPair on 0xc00000ec08
DEBU[0000] UsbMux Receive on 0xc00000ec38
DEBU[0000] UsbMux sendios.connectMessage on 0xc00000ec08
DEBU[0000] UsbMux Receive on 0xc0001e4890
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc000353240 and wrap with tlsConn0xc0004c4058
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=true Port=50401 Request=StartService Service=com.apple.mobile.house_arrest
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000353240
DEBU[0000] Opening connection:0xc0004ae0c0
DEBU[0000] UsbMux sendios.ReadPair on 0xc0003e40e0
DEBU[0000] UsbMux Receive on 0xc0003e4110
DEBU[0000] Opening connection:0xc00042a2b0
DEBU[0000] UsbMux sendios.connectMessage on 0xc00000ebd8
DEBU[0000] UsbMux Receive on 0xc00000ec20
DEBU[0000] enable session ssl on0xc000352be0 and wrap with tlsConn0xc0004c4018
DEBU[0000] Lockdown sendmap[string]interface {}
DEBU[0000] Closing connection:0xc00010f780
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x135cafe]

goroutine 6 [running]:
github.com/danielpaulus/go-ios/ios/house_arrest.Connection.Close({{0x0, 0x0}, 0xc0000b5208})
	/Users/runner/work/go-ios/go-ios/ios/house_arrest/house_arrest.go:84 +0x1e
github.com/danielpaulus/go-ios/ios/testmanagerd.setupXcuiTest({0xf, {0xc00035f640, 0x8}, {0x1c9c3800, {0xc00035f710, 0x3}, 0xf, 0x14100000, 0x12a8, {0xc00002f7a0, ...}}}, ...)
	/Users/runner/work/go-ios/go-ios/ios/testmanagerd/xcuitestrunner.go:490 +0x88c
github.com/danielpaulus/go-ios/ios/testmanagerd.runXUITestWithBundleIdsXcode12({0x7ffeefbff9f8, 0x0}, {0x7ffeefbffa37, 0x29}, {0x7ffeefbffa70, 0x1b}, {0xf, {0xc00035f640, 0x8}, {0x1c9c3800, ...}}, ...)
	/Users/runner/work/go-ios/go-ios/ios/testmanagerd/xcuitestrunner.go:240 +0xde
github.com/danielpaulus/go-ios/ios/testmanagerd.RunXCUIWithBundleIds({0x7ffeefbff9f8, 0x29}, {0x7ffeefbffa37, 0x29}, {0x7ffeefbffa70, 0x1b}, {0xf, {0xc00035f640, 0x8}, {0x1c9c3800, ...}}, ...)
	/Users/runner/work/go-ios/go-ios/ios/testmanagerd/xcuitestrunner.go:310 +0x10fc
main.Main.func1()
	/Users/runner/work/go-ios/go-ios/main.go:438 +0xda
created by main.Main
	/Users/runner/work/go-ios/go-ios/main.go:437 +0x1ae5
```

phone: IPhone 6S on 14.7.1
go-ios version: [v1.0.34](https://github.com/danielpaulus/go-ios/releases/download/v1.0.34/go-ios-mac.zip)
xcode version: 12.5.1

**Comment by 0x3c3e** [comment]
In case of renaming com.0x3c3e.WebDriverAgentRunner.xctrunner to com.facebook.WebDriverAgentRunner.xctrunner it looks the same
```
➜  /tmp ./ios runwda -t --nojson
INFO[0000] Set Trace mode
DEBU[0000] map[--arg:[] --basedir:<nil> --binary:false --bundleid:<nil> --details:false --env:[] --help:false --nojson:true --output:<nil> --p12file:<nil> --password:<nil> --path:<nil> --pid:<nil> --plist:false --process:<nil> --setlang:<nil> --setlocale:<nil> --stop-at-entry:false --system:false --testrunnerbundleid:<nil> --trace:true --udid:<nil> --verbose:false --version:false --xctestconfig:<nil> <app_path>:<nil> <bundleID>:<nil> <hostPort>:<nil> <key>:[] <profileId>:<nil> <profileTypeId>:<nil> <targetPort>:<nil> apps:false auto:false ax:false date:false debug:false devicename:false devicestate:false diagnostics:false dproxy:false enable:false forward:false image:false info:false install:false kill:false lang:false launch:false list:false listen:false mobilegestalt:false mount:false pair:false pcap:false ps:false readpair:false reboot:false runtest:false runwda:true screenshot:false syslog:false version:false]
DEBU[0000] Looking for device ''
DEBU[0000] Opening connection:0xc00034ed80
DEBU[0000] UsbMux sendios.ReadDevicesType on 0xc00000e8a8
DEBU[0000] UsbMux Receive on 0xc00000e8f0
DEBU[0000] no udid specified using default               udid=e07968a501322f6e52993ed13d1eb34a6288b463
DEBU[0000] Opening connection:0xc00034f0d0
DEBU[0000] UsbMux sendios.ReadPair on 0xc00000f088
DEBU[0000] UsbMux Receive on 0xc00000f0d0
DEBU[0000] UsbMux sendios.connectMessage on 0xc00000f088
DEBU[0000] UsbMux Receive on 0xc00000f820
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc000352d20 and wrap with tlsConn0xc00032b2d8
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=true Port=50556 Request=StartService Service=com.apple.mobile.diagnostics_relay
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000352d20
DEBU[0000] Opening connection:0xc00034fa40
DEBU[0000] UsbMux sendios.ReadPair on 0xc0003da1e8
DEBU[0000] UsbMux Receive on 0xc0003da218
DEBU[0000] Opening connection:0xc00034fbe0
DEBU[0000] UsbMux sendios.connectMessage on 0xc0003da920
DEBU[0000] UsbMux Receive on 0xc0003da950
DEBU[0000] enable session ssl on0xc0003c8c00 and wrap with tlsConn0xc00032b2f8
INFO[0000] no bundle ids specified, falling back to defaults
INFO[0000] Running wda                                   bundleid=com.facebook.WebDriverAgentRunner.xctrunner testbundleid=com.facebook.WebDriverAgentRunner.xctrunner xctestconfig=WebDriverAgentRunner.xctest
DEBU[0000] Opening connection:0xc000098000
DEBU[0000] UsbMux sendios.ReadPair on 0xc0000ae020
DEBU[0000] UsbMux Receive on 0xc0000ae050
DEBU[0000] UsbMux sendios.connectMessage on 0xc0000ae020
DEBU[0000] UsbMux Receive on 0xc0000ae758
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc000096000 and wrap with tlsConn0xc00009e010
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=true Port=50559 Request=StartService Service=com.apple.testmanagerd.lockdown.secure
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000096000
DEBU[0000] Opening connection:0xc000596200
DEBU[0000] UsbMux sendios.ReadPair on 0xc0003da3f8
DEBU[0000] UsbMux Receive on 0xc0003da428
DEBU[0000] Opening connection:0xc0005963a0
DEBU[0000] UsbMux sendios.connectMessage on 0xc0003dab30
DEBU[0000] UsbMux Receive on 0xc0003dab60
DEBU[0000] enable session ssl on0xc0003c8ce0 and wrap with tlsConn0xc000010028
DEBU[0000] Opening connection:0xc000068850
DEBU[0000] UsbMux sendios.ReadPair on 0xc000440050
DEBU[0000] UsbMux Receive on 0xc000440080
DEBU[0000] UsbMux sendios.connectMessage on 0xc000440050
DEBU[0000] Dispatching:_notifyOfPublishedCapabilities:
DEBU[0000] Global Dispatcher Received: [_notifyOfPublishedCapabilities:] [{t:binary, v:{"com.apple.private.DTXBlockCompression":2,"com.apple.private.DTXConnection":1}},]
DEBU[0000] UsbMux Receive on 0xc000440788
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc000352000 and wrap with tlsConn0xc000010030
DEBU[0000] Lockdown sendios.valueRequest
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000352000
DEBU[0000] Opening connection:0xc000596a90
DEBU[0000] UsbMux sendios.ReadPair on 0xc0003db070
DEBU[0000] UsbMux Receive on 0xc0003db0a0
DEBU[0000] UsbMux sendios.connectMessage on 0xc0003db070
DEBU[0000] UsbMux Receive on 0xc0000ae830
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc0003c9420 and wrap with tlsConn0xc00009e020
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=false Port=50563 Request=StartService Service=com.apple.mobile.installation_proxy
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc0003c9420
DEBU[0000] Opening connection:0xc000596c30
DEBU[0000] UsbMux sendios.ReadPair on 0xc0003db2e0
DEBU[0000] UsbMux Receive on 0xc0003db310
DEBU[0000] Opening connection:0xc0000986d0
DEBU[0000] UsbMux sendios.connectMessage on 0xc0000af568
DEBU[0000] UsbMux Receive on 0xc0000af598
DEBU[0000] Lockdown sendmap[string]interface {}
DEBU[0000] Opening connection:0xc000253050
DEBU[0000] UsbMux sendios.ReadPair on 0xc0001fbc40
DEBU[0000] UsbMux Receive on 0xc0001fbc70
DEBU[0000] UsbMux sendios.connectMessage on 0xc0001fbc40
DEBU[0000] UsbMux Receive on 0xc0000ae188
DEBU[0000] Lockdown sendios.startSessionRequest
DEBU[0000] enable session ssl on0xc000258140 and wrap with tlsConn0xc000446000
DEBU[0000] Lockdown sendios.startServiceRequest
DEBU[0000] Service started on device                     EnableServiceSSL=true Port=50566 Request=StartService Service=com.apple.mobile.house_arrest
DEBU[0000] Lockdown sendios.stopSessionRequest
DEBU[0000] Closing connection:0xc000258140
DEBU[0000] Opening connection:0xc000252250
DEBU[0000] UsbMux sendios.ReadPair on 0xc0000ae4e8
DEBU[0000] UsbMux Receive on 0xc0000ae518
DEBU[0000] Opening connection:0xc000068910
DEBU[0000] UsbMux sendios.connectMessage on 0xc00000e860
DEBU[0000] UsbMux Receive on 0xc00000e890
DEBU[0000] enable session ssl on0xc000072920 and wrap with tlsConn0xc000010030
DEBU[0000] Lockdown sendmap[string]interface {}
DEBU[0000] Closing connection:0xc000097900
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x135c8fe]

goroutine 6 [running]:
github.com/danielpaulus/go-ios/ios/house_arrest.Connection.Close({{0x0, 0x0}, 0xc000125068})
	/Users/runner/work/go-ios/go-ios/ios/house_arrest/house_arrest.go:84 +0x1e
github.com/danielpaulus/go-ios/ios/testmanagerd.setupXcuiTest({0xf, {0xc00034c500, 0x8}, {0x1c9c3800, {0xc00034c5d0, 0x3}, 0xf, 0x14100000, 0x12a8, {0xc000021710, ...}}}, ...)
	/Users/runner/work/go-ios/go-ios/ios/testmanagerd/xcuitestrunner.go:490 +0x88c
github.com/danielpaulus/go-ios/ios/testmanagerd.runXUITestWithBundleIdsXcode12({0x142be96, 0x0}, {0x142be96, 0x2b}, {0x1422a4d, 0x1b}, {0xf, {0xc00034c500, 0x8}, {0x1c9c3800, ...}}, ...)
	/Users/runner/work/go-ios/go-ios/ios/testmanagerd/xcuitestrunner.go:240 +0xde
github.com/danielpaulus/go-ios/ios/testmanagerd.RunXCUIWithBundleIds({0x142be96, 0x2b}, {0x142be96, 0x2b}, {0x1422a4d, 0x1b}, {0xf, {0xc00034c500, 0x8}, {0x1c9c3800, ...}}, ...)
	/Users/runner/work/go-ios/go-ios/ios/testmanagerd/xcuitestrunner.go:310 +0x10fc
main.Main.func1()
	/Users/runner/work/go-ios/go-ios/main.go:430 +0xda
created by main.Main
	/Users/runner/work/go-ios/go-ios/main.go:429 +0x1ac5
```

**Comment by danielpaulus** [comment]
Thanks for reporting this!
Should be fixed with https://github.com/danielpaulus/go-ios/pull/82

**Comment by danielpaulus** [comment]
should be resolved with #87
